commec/tools/database_handler.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2024 International Biosecurity and Biosafety Initiative for Science
+"""
+Abstract base class for a database handler. 
+Customize to screen desired database functionality.
+"""
+import os
+from dataclasses import dataclass
+import subprocess
+import logging
+
+@dataclass
+class DatabaseVersion():
+    """ Container class for outputting version related information from a database."""
+    def __init__(self, input_version : str = "x.x.x", input_date : str = "Null", input_comment :str = ""):
+        self.version_string = input_version
+        self.version_date = input_date
+        self.additional_comment = input_comment
+    version_string : str
+    version_date : str
+    additional_comment : str
```

Comment:
Does it make sense to split into a tool / command version (e.g. "BLAST+ 2.15") and database version (which for BLAST databases we can pull out using blastdbcmd, but it's not semver, it's a build date https://www.ncbi.nlm.nih.gov/books/NBK569853/).

Reply:
Kept it simple with a string for tool and a string for database, and these can be filled however they like - at the moment all databases are programmed to return a short snippet from various cli's which I deemed would be useful.